From 857c48c43aa11895eaa7e8d90b2fdf2850c5892e Mon Sep 17 00:00:00 2001
From: iamara <iskander.amara.it@gmail.com>
Date: Mon, 22 Jul 2024 22:35:48 +0200
Subject: [PATCH] arm64: dts: ls2080a-rdb: Add definition of RJ45 ports phy
 nodes

As noticed in earlier dtbs for ls2080a rdb board, there are missing
definition of phy ports, therefore let's enable add them and enable
the correspondant mdio nodes.

Signed-off-by: iamara <iskander.amara.it@gmail.com>
Signed-off-by: consider it GmbH <innovation.support@consider-it.de>
---
 .../boot/dts/freescale/fsl-ls2080a-rdb.dts    | 24 ++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls2080a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls2080a-rdb.dts
index 1c8c99a74071..da16af1539c1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls2080a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls2080a-rdb.dts
@@ -24,6 +24,26 @@ chosen {
 	};
 };
 
+&dpmac1 {
+	phy-handle = <&mdio1_phy1>;
+	/delete-property/ pcs-handle;
+};
+
+&dpmac2 {
+	phy-handle = <&mdio1_phy2>;
+	/delete-property/ pcs-handle;
+};
+
+&dpmac3 {
+	phy-handle = <&mdio1_phy3>;
+	/delete-property/ pcs-handle;
+};
+
+&dpmac4 {
+	phy-handle = <&mdio1_phy4>;
+	/delete-property/ pcs-handle;
+};
+
 &dpmac5 {
 	phy-handle = <&mdio2_phy1>;
 	phy-connection-type = "10gbase-r";
@@ -45,7 +65,7 @@ &dpmac8 {
 };
 
 &emdio1 {
-	status = "disabled";
+	status = "okay";
 
 	/* CS4340 PHYs */
 	mdio1_phy1: emdio1_phy@1 {
@@ -66,6 +86,8 @@ mdio1_phy4: emdio1_phy@4 {
 };
 
 &emdio2 {
+	status = "okay";
+
 	/* AQR405 PHYs */
 	mdio2_phy1: emdio2_phy@1 {
 		compatible = "ethernet-phy-ieee802.3-c45";
-- 
2.43.0

